// react-leaflet-universal/dist/decorator.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = decorate;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var node = _propTypes.PropTypes.node,
    oneOfType = _propTypes.PropTypes.oneOfType,
    func = _propTypes.PropTypes.func,
    object = _propTypes.PropTypes.object;

function decorate(componentName) {
  var displayName = "LeafletUniv".concat(componentName);

  var Decorated =
  /*#__PURE__*/
  function (_Component) {
    (0, _inherits2["default"])(Decorated, _Component);

    function Decorated(props) {
      var _this;

      (0, _classCallCheck2["default"])(this, Decorated);
      _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(Decorated).call(this, props));
      _this.state = {
        loaded: false
      };
      _this.constructor.displayName = displayName;
      _this._leafletRef = _react["default"].createRef();

      _this._getRef = function () {
        return _this.props.leafletRef || _this._leafletRef;
      };

      return _this;
    }

    (0, _createClass2["default"])(Decorated, [{
      key: "componentDidMount",
      value: function componentDidMount() {
        this.setState(function () {
          return {
            loaded: true
          };
        });
        this.ClientComponent = require('react-leaflet')[componentName];
      }
    }, {
      key: "componentDidUpdate",
      value: function componentDidUpdate() {
        var ref = this._getRef();

        if (ref && ref.current && this.leafletElement !== ref.current.leafletElement) {
          this.leafletElement = ref.current.leafletElement;
        }
      }
    }, {
      key: "render",
      value: function render() {
        if (!this.state.loaded) return null;
        var ClientComponent = this.ClientComponent;
        var _this$props = this.props,
            children = _this$props.children,
            rest = (0, _objectWithoutProperties2["default"])(_this$props, ["children"]);
        var childComponents = typeof children === 'function' ? children() : children;
        return _react["default"].createElement(ClientComponent, (0, _extends2["default"])({}, rest, {
          ref: this._getRef()
        }), childComponents);
      }
    }]);
    return Decorated;
  }(_react.Component);

  Decorated.displayName = displayName;
  Decorated.propTypes = {
    children: oneOfType([node, func]),
    leafletRef: oneOfType([func, object])
  };
  return Decorated;
}

module.exports = exports.default;
